File: Java/6- JCF/src/Slicing/Slicing.java
package Slicing;

class Persona{
	String name;

	Persona () {}
	
	Persona(String name) {
		this.name = name;
	}
}


class Studente extends Persona {
	int cod;
	
	 Studente(String name, int cod) {
		super(name);
		this.cod = cod;
	}
}

public class Slicing {
	public static void main(String[] args) {
		Studente s = new Studente("marco", 1064889);
		
		Persona p = s;

		//Questo non si può fare -> errore di compilazione
		//System.out.println(p.cod);

		System.out.println(p.name);
	}
}

File: Cpp/prove/Slicing.cpp

#include <iostream>
using namespace std;

class Persona
{
public:
    void print() { cout << "print persona" << endl; }
    virtual void stampa() { cout << "stampa persona" << endl; }
};

class Studente : public Persona
{

public:
    string matricola;
    void print() { cout << "print studente" << endl; }
    virtual void stampa() { cout << "stampa studente " << this->matricola << endl; }
};

int main(int argc, char **argv)
{
    // Questo non si può fare
    // Persona p = Studente;
}